synapse/storage/roommember.py

```diff
@@ -16,16 +16,22 @@
 import logging
 from collections import namedtuple
 
+import attr
+
+from synapse.types import PersistedEventPosition
+
 logger = logging.getLogger(__name__)
 
 
 RoomsForUser = namedtuple(
     "RoomsForUser", ("room_id", "sender", "membership", "event_id", "stream_ordering")
 )
 
-GetRoomsForUserWithStreamOrdering = namedtuple(
-    "GetRoomsForUserWithStreamOrdering", ("room_id", "event_pos")
-)
+
+@attr.s(slots=True, frozen=True, auto_attribs=True)
```

Comment:
Could we add `weakref_slot=False` for things that are going to be cached a lot? That cuts out quite a few bytes.

Reply:
I put it on all of them since they're all cached. Hopefully that seems OK?

Reply:
Sounds good, there's no downside to it since we don't use weakrefs

Reply:
I think e97ac25282d763c438221d462ece249f0e21a757 added `weakref_slot=True` rather than `=False` --- don't follow that given the above

Reply:
Oh, you're right. I messed that up. 🤦 Thanks for catching it!

Reply:
See #10642

Reply:
I'm glad someone is on the ball today :slightly_smiling_face: